src/gen/conditions.rs

```diff
@@ -186,7 +186,7 @@ pub fn parse_args(
             // AGG_SIG_* take two parameters
 
             if (flags & STRICT_ARGS_COUNT) != 0 {
-                check_nil(a, c)?;
+                check_nil(a, rest(a, c)?)?;
```

Comment:
this is the bugfix. the nil-terminator is expected to be the *next* item in the list, not the last cons-cell.
The rest are extended tests to cover this and possible other differences in mempool-mode and consensus-mode